common/src/main/scala/uk/ac/wellcome/platform/archive/common/bagit/models/BagManifest.scala

```diff
@@ -0,0 +1,51 @@
+package uk.ac.wellcome.platform.archive.common.bagit.models
+
+import java.io.InputStream
+
+import uk.ac.wellcome.platform.archive.common.verify.{ChecksumValue, HashingAlgorithm}
+
+import scala.util.matching.Regex
+import scala.util.{Failure, Success, Try}
+
+case class BagManifest(
+                        checksumAlgorithm: HashingAlgorithm,
+                        files: List[BagFile]
+                      )
+
+object BagManifest {
+  val lineRegex: Regex = """(.+?)\s+(.+)""".r
+
+  def create(stream: InputStream, algorithm: HashingAlgorithm): Try[BagManifest] = {
+
+    val lines = scala.io.Source
+      .fromInputStream(stream)
+      .mkString
+      .split("\n")
+      .filter { _.nonEmpty }
+      .toList
```

Comment:
minor: This reads the whole string into memory first, I think, with `mkString`?

I was experimenting with ways to read lines from a stream that might be a tad more memory-efficient the other day – see https://notebook.alexwlchan.net/2019/05/scala-iterating-over-the-newlines-of-an-inputstream/ – but I don’t think it makes a huge amount of difference.